### PR TITLE
[Enhancement] Add default values for sparse_optimizer_lr and weight_decay

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR set default values for the sparse_optimizer_lr and weight_decay in Models' `init_optimizer()`. 

Reasons of this PR:
1. Most of users did not modify the two arguments values, rather than use values preset in our yaml files.
2. Set the two arguments will make API calls complex.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/configuration/configuration-run.rst [hunk 1]
```diff
@@ -185,7 +185,7 @@ GraphStorm provides a set of parameters to control training hyper-parameters.
 
     - Yaml: ``sparse_optimizer_lr: 0.5``
     - Argument: ``--sparse-optimizer-lr 0.5``
-    - Default value: Same as lr.
```
Comment (classicsong) [inline docs/source/configuration/configuration-run.rst:188]: Why change the default value from lr to 0.01?

File: python/graphstorm/model/gnn.py [hunk 2]
```diff
@@ -711,7 +711,7 @@ def restore_sparse_model(self, restore_model_path):
         # restore sparse embeddings for node_input_encoder.
         load_sparse_embeds(restore_model_path, self.node_input_encoder)
 
-    def init_optimizer(self, lr, sparse_optimizer_lr, weight_decay, lm_lr=None):
+    def init_optimizer(self, lr, sparse_optimizer_lr=0.01, weight_decay=0, lm_lr=None):
```
Comment (classicsong) [inline python/graphstorm/model/gnn.py:714]: You can set default value of sparse_optimizer_lr to None and check if it is None, use lr.

File: docs/source/configuration/configuration-run.rst (hunk 1 above)
Comment (zhjwy9343) [reply docs/source/configuration/configuration-run.rst:188]: revert to use lr for the sparse_optimizer_lr. thanks

File: python/graphstorm/model/gnn.py (hunk 2 above)
Comment (zhjwy9343) [reply python/graphstorm/model/gnn.py]: Agree. Changed.

File: python/graphstorm/model/node_glem.py [hunk 3]
```diff
@@ -156,10 +156,13 @@ def inference_route_is_gnn(self):
         # GLEM is being trained: use gnn route if not training lm
         return not self.training_lm
 
-    def init_optimizer(self, lr, sparse_optimizer_lr, weight_decay, lm_lr=None):
+    def init_optimizer(self, lr, sparse_optimizer_lr=None, weight_decay=0, lm_lr=None):
         """Initialize optimzer, which will be stored in self.lm._optimizer, self.gnn._optimizer
         """
         sparse_params = self.lm.get_sparse_params()
+        # check and set the sparse optimizer learning rate
+        if sparse_optimizer_lr is None:
```
Comment (jalencato) [inline python/graphstorm/model/node_glem.py:164]: The sparse_optimizer_lr should have been set to lr here: https://github.com/awslabs/graphstorm/blob/2bf5c404348a1ad7805150426fa5748f60e923aa/python/graphstorm/config/argument.py#L1169
Also same as the previous

File: python/graphstorm/model/gnn.py [hunk 4]
```diff
@@ -720,14 +720,17 @@ def init_optimizer(self, lr, sparse_optimizer_lr, weight_decay, lm_lr=None):
...
-            The weight decay for the optimizer.
+            The weight decay for the optimizer. Default is 0.0.
         lm_lr: float
             Language model fine-tuning learning rate for
             langauge model dense parameters.
         """
         sparse_params = self.get_sparse_params()
+        # check and set the sparse optimizer learning rate
+        if sparse_optimizer_lr is None:
+            sparse_optimizer_lr = lr
```
Comment (jalencato) [inline python/graphstorm/model/gnn.py:733]: Same here

Comment (zhjwy9343) [reply python/graphstorm/model/gnn.py:733]: updated

File: python/graphstorm/model/node_glem.py (hunk 3 above)
Comment (zhjwy9343) [reply python/graphstorm/model/node_glem.py:164]: updated.

**Review by classicsong — APPROVED** [review]

**Pull request merged**